examples/junit/src/test/java/com/example/AutofuzzFuzzTests.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.example;
+
+import com.code_intelligence.jazzer.api.FuzzerSecurityIssueLow;
+import com.code_intelligence.jazzer.junit.FuzzTest;
+
+class AutofuzzFuzzTests {
```

Comment:
I tried to add an inner class, to check the package visibility checks, and use that as test parameter, but directly get an error that the parameter is null. Could you please add a similar test case?

Reply:
Could you describe the failure in a bit more detail? I added a private static class and verified and the test passes.

Reply:
In an autofuzz fuzzing run the method parameter has a good chance of being `null`. Like in this example:

```
  static class InnerClass {
    private String str;

    InnerClass(String str) {
      this.str = str;
    }

    void validate() {
      if ("jazzer".equals(str)) {
        throw new RuntimeException();
      }
    }
  }

  @FuzzTest
  void innerClass(InnerClass innerClassParam) {
    // Ignore assumptions exceptions during fuzzing run?
    // assumeTrue(innerClassParam != null);
    // if (innerClassParam == null) {
    //   return;
    // }
    innerClassParam.validate();
  }
```

Of course null-checking the parameter works, but is that what users expect? On the other hand, providing `null`, from time to time, is also valuable.

I noticed that exceptions raised in assumptions let a fuzzing run fail, should we ignore those while fuzzing?

Reply:
I pushed a commit that ignores `TestAbortedException`. I really like `assumeTrue(innerClassParam != null)`. Given the chaos surrounding `@NonNull`-like annotations and the fact that it may be impossible for the fuzzer to generate a non-`null` value given its current input, I think that `assumeX` models this quite well and should be our preferred way to handle this situation.